### DISCUSSION TITLE
Using FBX sanitizer to check Sketchup FBX exports before Unity import

### DISCUSSION THREAD

**Discussion by Schmitt-DE** [Pontoco/fbx-sanitizer#3] [Ideas]
I want to use the FBX sanitizer to check Sketchup FBX exports before I import the FBX file in Unity. Sketchup exports FBX files as ASCII files (not binary). The FBX sanitizer fails as soon as it recognizes that the FBX files is non-binary but ASCII. Do you plan to handle ASCII files as well? It would be great if file format would be handled like any other error and not completely quit the application before showing the real relevant error that I need :-(

**Comment by Kleptine** [comment]
The reason FBX Sanitizer doesn't support ascii files is because the FBX parser it uses [fbxcel-dom](https://github.com/lo48576/fbxcel-dom) does not support them. 

The only implementation of the FBX Ascii format that I'm aware of is the Autodesk FBX SDK itself. We don't use that because:
 - It would require installing the SDK alongside the tool (it can't be statically compiled into a single exe).
 - The tool would become massive, many hundreds of MB of code. (The Autodesk SDK is huge)
 - The Autodesk FBX SDK doesn't support Rust. (Rust makes this tool much easier to maintain for cross platform)
 - Licensing and distribution may become complicated.

So for those reasons I used the fbxcel-dom FBX parser instead. 

It would unfortunately be a massive task to add ASCII support to fbxcel-dom, because it would require reverse-engineering the ascii format (there's no spec that I'm aware of). To give a sense of the difficulty, even major tools like Blender (that have a vested interested in industry support) do not support the FBX Ascii format.

My recommendation would be to find a way to convert the files. The [FBXConverter ](https://www.autodesk.com/developer-network/platform-technologies/fbx-converter-archives)tool is old, but still very functional. I use this all the time to convert between ascii and binary.

**Reply by Schmitt-DE** [reply]
Thank you very much for the explanation! Since it is obviously a huge task to implement ASCII functionality and there are probably only few users using Sketchup and Unity, since most user probably use blender/Maja => Unity workflow, just forget my feature request! I will instead check FBX ASCII to binary converters and THEN use the FBX Sanitizer.

**Comment by Kleptine** [comment]
I can improve the error message string for ASCII files, if it helps, but since we can't read the ascii files, there's no way to do anything other than exit.
